spherical_geometry/tests/test_basic.py

```diff
@@ -529,8 +529,28 @@ def test_math_util_angle_domain():
 
 
 def test_math_util_length_domain():
-    with pytest.raises(ValueError):
-        great_circle_arc.length([[np.nan, 0, 0]], [[0, 0, np.inf]])
+    a = [
+        [np.nan, 0, 0],
+        [np.nan, 0, 0],
+        [np.nan, np.nan, np.nan],
+        [0, 0, 0],
+        [0, 0, 0],
+        [0, 0, np.nan],
+        [0, 0, 0],
+    ]
+    b = [
+        [0, 0, np.inf],
+        [0, 0, np.inf],
+        [0, 0, 0],
+        [np.inf, np.inf, np.inf],
+        [0, 0, np.inf],
+        [0, 0, 0],
+        [0, 0, 0],
+    ]
+
+    v = great_circle_arc.length(a, b)
+
+    assert np.all(np.logical_not(np.isfinite(v)))
```

Comment:
To check I'm understanding. Previously providing `np.nan` to `length` resulted in `ValueError: Out of domain for acos` and with this PR it doesn't raise an exception and instead returns `np.nan`?

Where is `length` used and will this change in behavior have consequences?